docs/src/ref_impl/build_and_run.md

```diff
@@ -0,0 +1,104 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Build and Run
+
+The canonical URL for the upstream Ghaf git repository is <https://github.com/tiiuae/ghaf>. To try Ghaf, you can build it from the source.
+
+Our supported targets are the following:
+
+| Hardware         | Architecture     | Scope         |
+|---               |---               |---            |
+| Generic Intel    | x86              | QEMU (VM)     |
+| NVIDIA Jetson AGX Orin  | AArch64   | hardware      |
+| NXP i.MX 8QM-MEK | AArch64          | hardware      |
+
+Scope of target support is updated with development progress.
+
+## Building Ghaf Image
+
+Before you begin, follow the basic device-independent steps:
+
+* Clone the git repository <https://github.com/tiiuae/ghaf>.
+* Ghaf uses a Nix flake approach to build the framework targets, make sure to:
+  * Install Nix or full NixOS if needed: <https://nixos.org/download.html>.
+  * Enable flakes: <https://nixos.wiki/wiki/Flakes#Enable_flakes>.  
+    To see all Ghaf supported outputs, type `nix flake show`.
+  * Set up an AArch64 remote builder: <https://nixos.org/manual/nix/stable/advanced-topics/distributed-builds.html>.
+
+---
+
+
+### Running Ghaf Image for x86 VM (ghaf-host)
+
+Run the `nix run .#packages.x86_64-linux.vm` command.
+
+This creates `ghaf-host.qcow2` copy-on-write overlay disk image in your current directory. If you do unclean shutdown for the QEMU VM, you might get weird errors the next time you boot. Simply removing `ghaf-host.qcow2` should be enough. To cleanly shut down the VM, from the menu bar of the QEMU Window, click Machine and then Power Down.
+
+
+---
+
+### Building Ghaf Image for NVIDIA Jetson Orin AGX
+
+If you cloned the ghaf repository and installed an AArch64 remote builder, run the `nix build .#packages.aarch64-linux.nvidia-jetson-orin-release` command to build the release build.
```

Comment:
We actually should recommend `-debug` target here. 
`-release` builds and boots but there's very little one can do with it by default (and by definition).

Reply:
Should I change this `nix build .#packages.aarch64-linux.nvidia-jetson-orin-release` to this `nixos-rebuild --flake .#<target-device> --target-host root@<device-ip-or-hostname> --fast switch`?